config/v1/types_cluster_operator.go

```diff
@@ -124,6 +124,10 @@ const (
 	// The binary maintained by the operator (eg: openshift-apiserver for the openshift-apiserver-operator) may still be
 	// available, but the user intent cannot be fulfilled.
 	OperatorFailing ClusterStatusConditionType = "Failing"
+
+	// OperatorUpgradeable indicates whether or not the operator is in a state where it can accept a new payload.  When
+	// set to `False` the CVO will disallow moving to a new version.
+	OperatorUpgradeable ClusterStatusConditionType = "Upgradeable"
```

Comment:
Since we've already established that it's not necessary an "upgrade", and also because i still think there might be some value in using this to determine general readiness of an operator, i'd propose renaming it to `Ready`.

Reply:
> Since we've already established that it's not necessary an "upgrade", and also because i still think there might be some value in using this to determine general readiness of an operator, i'd propose renaming it to `Ready`.

You cannot have the CVO base its decision about whether or not to allow a version change upon the operator status for the same reason you cannot allow it make a decision based on operand status.  If you did, you would be unable to fix a broken operator with a version change.  You're describing a different condition.

The upgradeable condition will *block* an upgrade and it should only be set in a case when we cannot safely allow a version change to proceed.

Reply:
> You cannot have the CVO base its decision about whether or not to allow a version change upon the operator status for the same reason you cannot allow it make a decision based on operand status.

who's setting this "upgradeable" condition if not the operator?

Reply:
> > You cannot have the CVO base its decision about whether or not to allow a version change upon the operator status for the same reason you cannot allow it make a decision based on operand status.
> 
> who's setting this "upgradeable" condition if not the operator?

Assuming you intend for `Ready` to mean your operator is functional.  If an operator is `Ready==false`, then we should definitely allow upgrading that operator because otherwise it can be stuck broken.  If an operator is `Ready=true`, then we should definitely allow upgrading that operator because it is functioning correctly.  It's a condition unrelated to whether or not the CVO should allow an upgrade.

Reply:
How/when do you define that operator is not ready (`Ready=false`) but upgradable (`Upgradable=true`)? What's the algorithm/conditions behind that decision? Is it human made?

Reply:
> How/when do you define that operator is not ready (`Ready=false`) but upgradable (`Upgradable=true`)? What's the algorithm/conditions behind that decision? Is it human made?

I think we're looking at different conditions and I'd like to have this pull focus on `Upgradeable`.  It is an operator made choice.